### ISSUE TITLE
add_to_chat - Potential issues

### ISSUE THREAD

**Issue by Byrth** [Windower/Issues#697]
Someone reported that battlemod has been occasionally blocking all battle chat, but when he unloads it the chat starts appearing again.
https://github.com/Byrth/Lua-Byrth/issues/368

If battlemod fails to generate a message or can't interpret the action packet, it doesn't alter the message field and lets it pass through and be generated normally. Battlemod sets the message field of the action packet to be 0 only immediately after injecting an add_to_chat command. Thus, this probably indicates that add_to_chat is blocking up for some reason.

**Comment by KenshiDRK** [comment]
I was the one having this issue and I think this could be caused by Highlight, Never happens if I have highlight unloaded.

**Comment by Byrth** [comment]
Good to know. Does this still occur?

**Comment by KenshiDRK** [comment]
I haven not tried again, after the issue I just never used highlight again. I'll try using both again.

**Comment by Byrth** [comment]
Okay, thanks

**Comment by KenshiDRK** [comment]
This bug is still present, using highlight and battlemod together causes battle text to stop displaying.

**Comment by SnickySnacks** [comment]
I'm not able to reproduce this. Is it possible that you have changed a settings file that is causing highlight to blank battlemod's output? Perhaps a nickname.xml that is being too greedy?

I've tried it with highlight loaded first and with battlemod loaded first and both work as expected.

**Comment by KenshiDRK** [comment]
I have changed any files, except for the filters on battlemod, so dunno. it doesn't happen inmediately you need to be playing for sometime to get it to happen, but for me it happens sooner or later.

**Comment by SnickySnacks** [comment]
Given that highlight seems to be writing color codes directly into the chat log, it seems like it may be corrupting the log. I had a similar issue in battlemod when writing an invalid character into the log caused FFXI to lock up when I opened the log window up to full screen. This happens with highlight loaded, too.

I'll look into it and see if I can figure out what's happening, exactly.

**Comment by SnickySnacks** [comment]
Yeah, it's highlight doing it. It's not writing the color codes correctly and can corrupt the log. Because it rewrites battlemod's color codes, this makes the problem worse.  For example, with battlemod turned off:

Snicky and Bogg happily dance together.
Normal chat output:
1E011E017FFC536E69636B797FFB20616E6420426F67672068617070696C792064616E636520746F6765746865722E7F
......Snicky.. and Bogg happily dance together..

Highlighted chat output:
1E011E017FFC1F081F081F081F081F081F081F08536E69636B791E011E011E011E011E011E011E017FFB20616E64201F081F081F081F081F081F081F08426F67671E011E011E011E011E011E011E012068617070696C792064616E636520746F6765746865722E7F
....................Snicky................ and ..............Bogg.............. happily dance together..

Not sure where the error in logic is happening, but it's highlight doing it, not battlemod.

**Comment by Byrth** [comment]
There's obviously something wrong with Highlight (hit 7 gsubs), but I'm not entirely sure the battlemod version is correct either. 1E01 is the color reset code, but it shouldn't necessarily lead the line. I assume the 7FFB and 7FFC are some kind of brackets around your name? They don't look like valid color codes to me.

I'll try to take a look at this when the campaign starts

**Comment by SnickySnacks** [comment]
the above is without battlemod loaded. What you're seeing is just whatever is being displayed normally. Battlemod color codes are the expected (color code)string(reset color) and seem fine, but you can see in the highlight lua that it attempts to undo the coloring on the battlemod lines.

Even without battlemod, highlight makes my chat log act weird (can't scroll back) and with battlemod it hard locks ffxi when trying to open the log.

I don't know enough about what it's doing to explain why the gsub seems like it's being called multiple times on each line of text.
